shared/js/background/chrome-events.es6.js

```diff
@@ -52,6 +52,25 @@ chrome.webRequest.onHeadersReceived.addListener(
     }
 )
 
+// keep track of URLs that the browser navigates to.
+//
+// this is currently meant to supplement tabManager.updateTabUrl() above:
+// tabManager.updateTabUrl only fires when a tab has finished loading with a 200,
+// which misses a couple of edge cases like browser special pages
+// and Gmail's weird redirect which returns a 200 via a service worker
+chrome.webNavigation.onCommitted.addListener(
+    (details) => {
+        // ignore navigation on iframes
+        if (details.frameId !== 0) return
+
+        const tab = tabManager.get({tabId: details.tabId})
+
+        if (!tab) return
+
+        tab.updateSite(details.url)
+    }
+)
```

Comment:
I was wondering whether, now that we have this, keeping the URL updated via `onHeadersReceived` was still necessary. We do still need to get the status code from that listener though.

Reply:
Really good call using onCommitted! I hadn't heard of it before, but looking at the docs it seems to be perfect for this case.

I suspect we can probably remove updating the url from `onHeadersReceived`, although I do wonder if that might expose some underlying race conditions since `onCommitted` occurs so much later in the cycle. I need to test this without the url updating in `onHeadersReceived` a bit more extensively, but as long as I don't run into any oddities there I would be pro removing it. One less check that has to be made on every request makes me happy.